### PR TITLE
Collective pallet: max proposal weight

### PR DESCRIPTION
**Pull request by muharem**
Introducing a limit for the weight of the proposed call.

polkadot companion: https://github.com/paritytech/polkadot/pull/6983
cumulus companion: https://github.com/paritytech/cumulus/pull/2410

### CODE REVIEW

**Comment by muharem** [comment]
bot rebase

File: frame/collective/src/lib.rs [hunk 1]
```diff
@@ -710,6 +714,11 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 	) -> Result<(u32, DispatchResultWithPostInfo), DispatchError> {
 		let proposal_len = proposal.encoded_size();
 		ensure!(proposal_len <= length_bound as usize, Error::<T, I>::WrongProposalLength);
+		let proposal_weight = proposal.get_dispatch_info().weight;
+		ensure!(
+			proposal_weight.all_lte(T::MaxProposalWeight::get()),
+			Error::<T, I>::WrongProposalWeight
+		);
```
Comment (bkchr) [inline frame/collective/src/lib.rs:721]: The proposal weight is already added to the weight of `propose` if `threshold < 2`. So, we would not even add this tx to a block when the weight is too big. TLDR: This should not be required.

Comment (bkchr) [reply frame/collective/src/lib.rs:721]: But yeah, to have the properties as `do_propose_proposed` it makes sense.

**Review by bkchr — APPROVED** [review]
Personally I would probably favor a `disapprove_proposal` that can be initiated by a simple majority of collective members to cancel a call.

**Comment by muharem** [comment]
bot merge

**Comment by muharem** [comment]
bot rebase

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2700058

**Comment by muharem** [comment]
bot merge

**Comment by ggwpez** [comment]
bot merge

**Pull request merged**